SwiftyUserDefaults 5.0.0-beta.2 cause error on Xcode 11 SwiftUI Canvas

I am using Xcode 11 GM Seed with SwiftyUserDefaults 5.0.0-beta.2, SwiftUI's canvas for previewing layout cause the following error.
I also tired to downgrade SwiftyUserDefaults version to 4.0.0 and everything works fine.





linker command failed with exit code 1 (use -v to see invocation)

----------------------------------------

Failed to build the scheme "Testing": Link /Users/daniel/Library/Developer/Xcode/DerivedData/Testing-epcackvmcahjtubjqjwkhgqsrspu/Build/Intermediates.noindex/Previews/Testing/Products/Debug-iphonesimulator/libCwlCatchException.dylib:
Undefined symbols for architecture x86_64:
  "_catchExceptionOfKind", referenced from:
      CwlCatchException.(catchReturnTypeConverter in _841BEC3F94220F0DFD4EA4E9DDB3DC93)<A where A: __C.NSException>(_: A.Type, block: () -> ()) -> A? in CwlCatchException.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)


Link /Users/daniel/Library/Developer/Xcode/DerivedData/Testing-epcackvmcahjtubjqjwkhgqsrspu/Build/Intermediates.noindex/Previews/Testing/Products/Debug-iphonesimulator/libCwlPreconditionTesting.dylib:
clang: error: no such file or directory: '/Users/daniel/Library/Developer/Xcode/DerivedData/Testing-epcackvmcahjtubjqjwkhgqsrspu/Build/Intermediates.noindex/Previews/Testing/Products/Debug-iphonesimulator/libCwlCatchException.dylib'
Command Ld failed with a nonzero exit code

--- a/.travis.yml
+++ b/.travis.yml
@@ -135,7 +135,7 @@ script:
   # - xcodebuild -project "$PROJECT" -scheme 'SwiftyUserDefaults' ONLY_ACTIVE_ARCH=NO CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO test | xcpretty
   
   - swift build
-  - swift test
+  - TEST=1 swift test
   - pod lib lint
 
   - scripts/cocoapods_integration_step.sh

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,7 @@
 ### Next
 
+* Fixed SPM integration so it no longer fetches testing libraries & doesn't create runtime crashes or Xcode Preview crashes anymore. [@sunshinejr](https://github.com/sunshinejr).
+
 ### 5.0.0-beta.2 (2019-09-09)
 
 * Added extensions for `Combine`! If you can `import Combine` and use it, check the `publisher(for:)` method on `DefaultsAdapter`. [@sunshinejr](https://github.com/sunshinejr)

--- a/Package.swift
+++ b/Package.swift
@@ -2,6 +2,26 @@
 // The swift-tools-version declares the minimum version of Swift required to build this package.
 
 import PackageDescription
+import class Foundation.ProcessInfo
+
+let shouldTest = ProcessInfo.processInfo.environment["TEST"] == "1"
+
+func resolveDependencies() -> [Package.Dependency] {
+    guard shouldTest else { return [] }
+
+    return [
+        .package(url: "https://github.com/Quick/Quick.git", .upToNextMajor(from: "2.0.0")),
+        .package(url: "https://github.com/Quick/Nimble.git", .upToNextMajor(from: "8.0.0")),
+    ]
+}
+
+func resolveTargets() -> [Target] {
+    let baseTarget = Target.target(name: "SwiftyUserDefaults", dependencies: [], path: "Sources")
+    let testTarget = Target.testTarget(name: "SwiftyUserDefaultsTests", dependencies: ["SwiftyUserDefaults", "Quick", "Nimble"])
+
+    return shouldTest ? [baseTarget, testTarget] : [baseTarget]
+}
+
 
 let package = Package(
     name: "SwiftyUserDefaults",
@@ -13,17 +33,6 @@ let package = Package(
             name: "SwiftyUserDefaults",
             targets: ["SwiftyUserDefaults"]),
     ],
-    dependencies: [
-        .package(url: "https://github.com/Quick/Quick.git", .upToNextMajor(from: "2.0.0")),
-        .package(url: "https://github.com/Quick/Nimble.git", .upToNextMajor(from: "8.0.0")),
-    ],
-    targets: [
-        .target(
-            name: "SwiftyUserDefaults",
-            dependencies: [],
-            path: "Sources"),
-        .testTarget(
-            name: "SwiftyUserDefaultsTests",
-            dependencies: ["SwiftyUserDefaults", "Quick", "Nimble"]),
-    ]
+    dependencies: resolveDependencies(),
+    targets: resolveTargets(),
 )

--- a/Package.swift
+++ b/Package.swift
@@ -34,5 +34,5 @@ let package = Package(
             targets: ["SwiftyUserDefaults"]),
     ],
     dependencies: resolveDependencies(),
-    targets: resolveTargets(),
+    targets: resolveTargets()
 )

--- a/Package@swift-4.2.swift
+++ b/Package@swift-4.2.swift
@@ -2,6 +2,26 @@
 // The swift-tools-version declares the minimum version of Swift required to build this package.
 
 import PackageDescription
+import class Foundation.ProcessInfo
+
+let shouldTest = ProcessInfo.processInfo.environment["TEST"] == "1"
+
+func resolveDependencies() -> [Package.Dependency] {
+    guard shouldTest else { return [] }
+
+    return [
+        .package(url: "https://github.com/Quick/Quick.git", .upToNextMajor(from: "1.3.0")),
+        .package(url: "https://github.com/Quick/Nimble.git", .upToNextMajor(from: "7.1.0"))
+    ]
+}
+
+func resolveTargets() -> [Target] {
+    let baseTarget = Target.target(name: "SwiftyUserDefaults", dependencies: [], path: "Sources")
+    let testTarget = Target.testTarget(name: "SwiftyUserDefaultsTests", dependencies: ["SwiftyUserDefaults", "Quick", "Nimble"])
+
+    return shouldTest ? [baseTarget, testTarget] : [baseTarget]
+}
+
 
 let package = Package(
     name: "SwiftyUserDefaults",
@@ -10,18 +30,7 @@ let package = Package(
             name: "SwiftyUserDefaults",
             targets: ["SwiftyUserDefaults"]),
     ],
-    dependencies: [
-        .package(url: "https://github.com/Quick/Quick.git", .upToNextMajor(from: "1.3.0")),
-        .package(url: "https://github.com/Quick/Nimble.git", .upToNextMajor(from: "7.1.0"))
-    ],
-    targets: [
-        .target(
-            name: "SwiftyUserDefaults",
-            dependencies: [],
-            path: "Sources"),
-        .testTarget(
-            name: "SwiftyUserDefaultsTests",
-            dependencies: ["SwiftyUserDefaults", "Quick", "Nimble"]),
-    ],
+    dependencies: resolveDependencies(),
+    targets: resolveTargets(),
     swiftLanguageVersions: [.v4_2]
 )

